### ISSUE TITLE
gexf parser conflict with gephi importer

### ISSUE THREAD

**Issue by gdbassett** [networkx/networkx#947]
I have an issue importing gexf files from networkx into gephi.  The issue seems to be around duplicate attributes.  I don't know all the cases in which gephi ignores attributes, however it at least ignores exact matches as well as case-insensitive matches.  Here's a simple example of a list of attributes and the associated output"
```
    <attributes class="edge" mode="static">
      <attribute id="4" title="Relationship_Type" type="string" />
    </attributes>
    <attributes class="node" mode="static">
      <attribute id="0" title="color" type="string" />
      <attribute id="1" title="number" type="integer" />
      <attribute id="2" title="Color" type="string" />
      <attribute id="3" title="Candy" type="integer" />
    </attributes>
```

Here's the error:
```
Default edge type set as UNDIRECTED
GEXF version 1.1 (deprecated)
Edge attribute found 'Relationship_Type' (STRING)
Node attribute found 'color' (STRING)
Node attribute found 'number' (INT)
Attribute with id '2' already exists, the attribute is ignored
Node attribute found 'Candy' (INT)
```

Here's a more complex example:
```
    <attributes class="edge" mode="static">
      <attribute id="26" title="Relationship_Type" type="string" />
      <attribute id="27" title="totalBytes" type="double" />
      <attribute id="28" title="totalFlows" type="double" />
    </attributes>
    <attributes class="edge" mode="dynamic">
      <attribute id="22" title="Relationship_Type" type="string" />
      <attribute id="23" title="totalBytes" type="double" />
      <attribute id="24" title="start" type="string" />
      <attribute id="25" title="totalFlows" type="integer" />
    </attributes>
    <attributes class="node" mode="dynamic">
      <attribute id="12" title="bytesInAsClient" type="integer" />
      <attribute id="13" title="bytesOutAsServer" type="integer" />
      <attribute id="14" title="bytesOutAsClient" type="integer" />
      <attribute id="15" title="start" type="string" />
      <attribute id="16" title="DNS" type="string" />
      <attribute id="17" title="flowsAsServer" type="integer" />
      <attribute id="18" title="model" type="string" />
      <attribute id="19" title="flowsAsClient" type="integer" />
      <attribute id="20" title="Class" type="string" />
      <attribute id="21" title="bytesInAsServer" type="integer" />
    </attributes>
    <attributes class="node" mode="static">
      <attribute id="0" title="coid" type="string" />
      <attribute id="1" title="DNS" type="string" />
      <attribute id="2" title="ip" type="string" />
      <attribute id="3" title="model" type="string" />
      <attribute id="4" title="cidr" type="string" />
      <attribute id="5" title="Class" type="string" />
      <attribute id="6" title="bytesInAsClient" type="integer" />
      <attribute id="7" title="bytesOutAsServer" type="integer" />
      <attribute id="8" title="bytesOutAsClient" type="integer" />
      <attribute id="9" title="flowsAsServer" type="integer" />
      <attribute id="10" title="flowsAsClient" type="integer" />
      <attribute id="11" title="bytesInAsServer" type="integer" />
    </attributes>
```

Here are the gephi errors:
```
Default edge type set as UNDIRECTED
GEXF version 1.1 (deprecated)
Edge attribute found 'Relationship_Type' (STRING)
Edge attribute found 'totalBytes' (DOUBLE)
Edge attribute found 'totalFlows' (DOUBLE)
Attribute with id '22' already exists, the attribute is ignored
Attribute with id '23' already exists, the attribute is ignored
Edge attribute found 'start' (DYNAMIC_STRING)
Attribute with id '25' already exists, the attribute is ignored
Node attribute found 'bytesInAsClient' (DYNAMIC_INT)
Node attribute found 'bytesOutAsServer' (DYNAMIC_INT)
Node attribute found 'bytesOutAsClient' (DYNAMIC_INT)
Node attribute found 'start' (DYNAMIC_STRING)
Node attribute found 'DNS' (DYNAMIC_STRING)
Node attribute found 'flowsAsServer' (DYNAMIC_INT)
Node attribute found 'model' (DYNAMIC_STRING)
Node attribute found 'flowsAsClient' (DYNAMIC_INT)
Node attribute found 'Class' (DYNAMIC_STRING)
Node attribute found 'bytesInAsServer' (DYNAMIC_INT)
Node attribute found 'coid' (STRING)
Attribute with id '1' already exists, the attribute is ignored
Node attribute found 'ip' (STRING)
Attribute with id '3' already exists, the attribute is ignored
Node attribute found 'cidr' (STRING)
Attribute with id '5' already exists, the attribute is ignored
Attribute with id '6' already exists, the attribute is ignored
Attribute with id '7' already exists, the attribute is ignored
Attribute with id '8' already exists, the attribute is ignored
Attribute with id '9' already exists, the attribute is ignored
Attribute with id '10' already exists, the attribute is ignored
Attribute with id '11' already exists, the attribute is ignored
Time Format: DOUBLE
```

I don't know if this a networkx thing, a gephi thing, or a gexf thing however it causes a significant number of issues with importing a graph that has nodes added in different way sat different times.

(Also, ignore the dynamic_int's.  Those are caused by gephi seeing the 'start' time attribute and consequently time phasing a bunch of the attributes.)

**Comment by mbastian** [comment]
Hi, this was a Gephi bug and has now been fixed. The patch will be included in the upcoming 0.9 release (planned for 12/20). Thanks for reporting

**Comment by mbastian** [comment]
And BTW, we're working on some adjustments on the GEXF format (version 1.3). You can see the discussion here: https://github.com/gephi/gephi/issues/1152. Feel free to chime in if there's something that would make the interconnection between networkx and gephi easier. Much appreciated!
